src/main/scala/inox/ast/SymbolOps.scala

```diff
@@ -1364,7 +1364,7 @@ trait SymbolOps { self: TypeOps =>
 
     symbols.instantiation_>:(formalType, actualType) match {
       case Some(tmap) =>
-        FunctionInvocation(fd.id, fd.tparams map { tpd => tmap.getOrElse(tpd.tp, tpd.tp) }, args)
+        FunctionInvocation(fd.id, fd.tparams map { tpd => tmap.getOrElse(tpd.tp, tpd.tp) }, args).setPos(fd.getPos)
```

Comment:
I hadn't noticed this but I don't think setting the position of the function to a call site makes much sense.
It would be better to call `copiedFrom` at the place where `functionInvocation` was called.